### PR TITLE
Edit bootstrapping chapter

### PR DESCRIPTION
**Pull request by pierwill**
Mainly revises the stages overview, using paragraphs instead of bullet points.

### CODE REVIEW

File: src/building/bootstrapping.md [hunk 1]
```diff
@@ -21,45 +18,61 @@ rustc, then uses it to compile the new compiler.
...
+and its 'target' or 'object' libraries (`std` and `rustc`).
+Both are staged, but in a staggered manner.
+
+### Stage 1
+
+The code in your clone is then
+compiled with the stage0 compiler to produce the stage1 compiler.
+However, since it was built with an older compiler (stage0),
+it was not compiled with any new optimizations in compiler source code.
+To optimize the stage1 compiler we go to the next stage.
```
[the remark this answers is not in the document]
Comment (pierwill) [reply src/building/bootstrapping.md:43]: I think as long as we don't say "optimizations are the reason for this stage", mentioning them here as an example gives an  unfamiliar reader an easy-to-understand example.

[the remark this answers is not in the document]
Comment (pierwill) [reply src/building/bootstrapping.md:43]: So `To optimize the stage1 compiler we go to the next stage` should be removed, yes.

File: src/building/bootstrapping.md [hunk 2]
```diff
@@ -21,45 +18,53 @@ rustc, then uses it to compile the new compiler.
...
+The stage0 compiler is then used only to compile `rustbuild`, `std`, and `rustc`.
+When compiling `rustc`, the stage0 compiler uses the freshly compiled `std`.
+There are two concepts at play here:
+a compiler (with its set of dependencies)
+and its 'target' or 'object' libraries (`std` and `rustc`).
+Both are staged, but in a staggered manner.
+
+### Stage 1
+
+The source code is then compiled with the stage0 compiler to produce the stage1 compiler.
```
Comment (jyn514) [inline src/building/bootstrapping.md:39]:
```suggestion
The rustc source code is then compiled with the stage0 compiler to produce the stage1 compiler.
```

File: src/building/bootstrapping.md [hunk 3]
```diff
@@ -174,20 +180,19 @@ Build artifacts include, but are not limited to:
   library/std` instead, which allows compiling programs without needing to define
   lang items.
 
-### Building vs. Running
-
+### Building vs. running
 
 Note that `build --stage N compiler/rustc` **does not** build the stage N compiler:
-instead it builds the stage _N+1_ compiler _using_ the stage N compiler.
+instead it builds the stage N+1_ compiler _using_ the stage N compiler.
```
Comment (jyn514) [inline src/building/bootstrapping.md:186]:
```suggestion
instead it builds the stage N+1 compiler _using_ the stage N compiler.
```

File: src/building/bootstrapping.md [hunk 4]
```diff
@@ -174,20 +180,19 @@ Build artifacts include, but are not limited to:
...
 In short, _stage 0 uses the stage0 compiler to create stage0 artifacts which
 will later be uplifted to be the stage1 compiler_.
 
 In each stage, two major steps are performed:
 
 1. `std` is compiled by the stage N compiler.
-2. That `std` is linked to programs built by the stage N compiler, including
-   the stage N artifacts (stage (N+1) compiler).
+2. That `std` is linked to programs built by the stage N compiler,
+   including the stage N artifacts <!-- (stage (N+1) compiler) -->.
```
Comment (jyn514) [inline src/building/bootstrapping.md:195]: I still think it makes sense to include this fragment.

File: src/building/bootstrapping.md [hunk 5]
```diff
@@ -202,16 +207,14 @@ Keep in mind this diagram is a simplification, i.e. `rustdoc` can be built at
...
-
 ### Stages and `std`
 
 Note that there are two `std` libraries in play here:
 1. The library _linked_ to `stageN/rustc`, which was built by stage N-1 (stage N-1 `std`)
 2. The library _used to compile programs_ with `stageN/rustc`, which was
    built by stage N (stage N `std`).
 
-Stage N `std` is pretty much necessary for any useful work with the stage N compiler.
+stage N `std` is pretty much necessary for any useful work with the stage N compiler.
```
Comment (jyn514) [inline src/building/bootstrapping.md:217]: This starts a sentence, I think "Stage" was correct.

**Comment by jyn514** [comment]
Thanks, this looks great! Do you mind squashing the commits a little? r=me after that :)

**Comment by pierwill** [comment]
Done.

r=jyn514

**Review by jyn514 — APPROVED** [review]

**Pull request merged**